import  org.junit.AfterClass;
import  org.junit.BeforeClass;
import  org.junit.Before;
import  org.junit.Test;
import static org.junit.Assert.*;
import  java.util.Map;
import  java.util.Arrays;
import  java.util.List;
import static org.evomaster.test.utils.EMTestUtils.*;
import  org.evomaster.client.java.controller.SutHandler;
import  io.restassured.RestAssured;
import static io.restassured.RestAssured.given;
import  io.restassured.response.ValidatableResponse;
import static org.hamcrest.Matchers.*;
import  io.restassured.config.JsonConfig;
import  io.restassured.path.json.config.JsonPathConfig;
import static io.restassured.config.RedirectConfig.redirectConfig;
import static org.evomaster.client.java.controller.contentMatchers.NumberMatcher.*;
import static org.evomaster.client.java.controller.contentMatchers.StringMatcher.*;
import static org.evomaster.client.java.controller.contentMatchers.SubStringMatcher.*;




/**
*  This file was automatically generated by EvoMaster on 2025-04-23T15:15:47.953230-04:00[America/Kentucky/Monticello]
 * <br>
*  The generated test suite contains 31 tests
 * <br>
*  Covered targets: 87
 * <br>
*  Used time: 1h 0m 0s
 * <br>
*  Needed budget for current results: 5%
 * <br>
*  
*/
public class EvoMaster_Test {

    
    private static String baseUrlOfSut = "http://localhost:12345/";
    
    
    @BeforeClass
    public static void initClass() {
        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
        RestAssured.useRelaxedHTTPSValidation();
        RestAssured.urlEncodingEnabled = false;
        RestAssured.config = RestAssured.config()
            .jsonConfig(JsonConfig.jsonConfig().numberReturnType(JsonPathConfig.NumberReturnType.DOUBLE))
            .redirect(redirectConfig().followRedirects(false));
    }
    
    
    
    
    
    
    
    
    /**
    * Calls:
    * (500) POST:/register
    * Found 2 potential faults. Type-codes: 100, 200
    */
    @Test(timeout = 60000)
    public void test_0_with500() throws Exception {
        
        // Fault100. HTTP Status 500. POST:/register
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. POST:/register -> Response status 500 not defined for path '/register'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" { " + 
                    " \"_links\": [ " + 
                    " { " + 
                    " \"deprecation\": \"Rr7z0bUo\", " + 
                    " \"href\": \"TMqEVFAkDiKne4G\", " + 
                    " \"hreflang\": \"7Eq\", " + 
                    " \"name\": \"PC1Yn6\", " + 
                    " \"rel\": { " + 
                    " \"EM_tainted_map\": \"_EM_2_XYZ_\" " + 
                    " }, " + 
                    " \"title\": \"VngfRnUeAxAdyg\" " + 
                    " }, " + 
                    " { " + 
                    " \"href\": \"azTuW3hTAgrfGJV\", " + 
                    " \"media\": \"AnO2nzo\", " + 
                    " \"profile\": \"htMv\", " + 
                    " \"title\": \"Li1vlqWYZLdPY\" " + 
                    " }, " + 
                    " { " + 
                    " \"deprecation\": \"P3T9j6_\", " + 
                    " \"media\": \"dwfGO3m\", " + 
                    " \"name\": \"oQNPU3LYf3PWO\", " + 
                    " \"type\": \"Of\" " + 
                    " }, " + 
                    " { " + 
                    " \"deprecation\": \"4V1IF2j7qVai\", " + 
                    " \"href\": \"RoKG3bjb3_gV3N\", " + 
                    " \"hreflang\": \"AWFraHYeg14\", " + 
                    " \"media\": \"GS\", " + 
                    " \"name\": \"9d25DszXAPA9u\", " + 
                    " \"rel\": { " + 
                    " \"EM_tainted_map\": \"_EM_2_XYZ_\" " + 
                    " }, " + 
                    " \"title\": \"637dStJpd\" " + 
                    " } " + 
                    " ], " + 
                    " \"email\": \"sechjss7M3y\", " + 
                    " \"name\": \"\\\\ \\\\\", " + 
                    " \"phone\": \"\" " + 
                    " } ")
                .post(baseUrlOfSut + "/register")
                .then()
                .statusCode(500)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("JSON parse error: Expected relation name; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Expected relation name\n at [Source: (PushbackInputStream); line: 1, column: 13] (through reference chain: market.dto.UserDTO[\"_links\"])"))
                .body("'description'", containsString("uri=/register"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (500) PUT:/customer/contacts
    * Found 2 potential faults. Type-codes: 100, 200
    */
    @Test(timeout = 60000)
    public void test_1_with500() throws Exception {
        
        // Fault100. HTTP Status 500. PUT:/customer/contacts
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/contacts -> Response status 500 not defined for path '/customer/contacts'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" { " + 
                    " \"_links\": [ " + 
                    " { " + 
                    " \"hreflang\": \"ci_D_mKsp9dr\", " + 
                    " \"name\": \"q\", " + 
                    " \"profile\": \"f1rh0dF\", " + 
                    " \"title\": \"GVqVg\" " + 
                    " }, " + 
                    " { " + 
                    " \"deprecation\": \"R\", " + 
                    " \"href\": \"PLuqzd\", " + 
                    " \"name\": \"UDTCS_xXY727\", " + 
                    " \"rel\": { " + 
                    " \"EM_tainted_map\": \"_EM_1_XYZ_\" " + 
                    " } " + 
                    " }, " + 
                    " { " + 
                    " \"deprecation\": \"cV28OdOCESss_p\", " + 
                    " \"hreflang\": \"KgYjoIqgiTq\", " + 
                    " \"profile\": \"g8Q5RgL\", " + 
                    " \"rel\": { " + 
                    " \"EM_tainted_map\": \"_EM_1_XYZ_\" " + 
                    " } " + 
                    " }, " + 
                    " { " + 
                    " \"name\": \"gYE_JSGo45aT\", " + 
                    " \"title\": \"mzOHLCpZrhoWeI\" " + 
                    " }, " + 
                    " { " + 
                    " \"deprecation\": \"E\", " + 
                    " \"href\": \"i5gu\", " + 
                    " \"media\": \"EZYeOrlXcU\", " + 
                    " \"profile\": \"GPkG4XwVnXKxLF8u\", " + 
                    " \"rel\": { " + 
                    " \"EM_tainted_map\": \"_EM_1_XYZ_\" " + 
                    " }, " + 
                    " \"type\": \"eTX4vzjtR4\" " + 
                    " } " + 
                    " ], " + 
                    " \"address\": \"eEqEhIWsW\" " + 
                    " } ")
                .put(baseUrlOfSut + "/customer/contacts")
                .then()
                .statusCode(500)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("JSON parse error: Expected relation name; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Expected relation name\n at [Source: (PushbackInputStream); line: 1, column: 13] (through reference chain: market.dto.ContactsDTO[\"_links\"])"))
                .body("'description'", containsString("uri=/customer/contacts"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (500) PUT:/customer/cart
    * Found 2 potential faults. Type-codes: 100, 200
    */
    @Test(timeout = 60000)
    public void test_2_with500() throws Exception {
        
        // Fault100. HTTP Status 500. PUT:/customer/cart
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/cart -> Response status 500 not defined for path '/customer/cart'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" { " + 
                    " \"_links\": [], " + 
                    " \"productId\": 51 " + 
                    " } ")
                .put(baseUrlOfSut + "/customer/cart")
                .then()
                .statusCode(500)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("JSON parse error: Expected relation name; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Expected relation name\n at [Source: (PushbackInputStream); line: 1, column: 13] (through reference chain: market.dto.CartItemDTO[\"_links\"])"))
                .body("'description'", containsString("uri=/customer/cart"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (500) PUT:/customer/cart
    * Found 2 potential faults. Type-codes: 100, 200
    */
    @Test(timeout = 60000)
    public void test_3_with500() throws Exception {
        
        // Fault100. HTTP Status 500. PUT:/customer/cart
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/cart -> Response status 500 not defined for path '/customer/cart'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" { " + 
                    " \"_links\": [ " + 
                    " { " + 
                    " \"deprecation\": \"WS2nQjejPK5WTD\", " + 
                    " \"hreflang\": \"UuxHykAiaF1ug\", " + 
                    " \"media\": \"iu2F6XmWJ9\", " + 
                    " \"profile\": \"3gVRC4jwfOAccWc\", " + 
                    " \"title\": \"g4M5j\" " + 
                    " }, " + 
                    " { " + 
                    " \"href\": \"C7VKMiAX\", " + 
                    " \"media\": \"Q58Zc\", " + 
                    " \"name\": \"sph8d\", " + 
                    " \"profile\": \"gv_yxEyzEhrsxEJU\", " + 
                    " \"title\": \"YOvfDD91UQT9\" " + 
                    " } " + 
                    " ], " + 
                    " \"quantity\": 576 " + 
                    " } ")
                .put(baseUrlOfSut + "/customer/cart?name=IUAYOIhYT")
                .then()
                .statusCode(500)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("JSON parse error: Expected relation name; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Expected relation name\n at [Source: (PushbackInputStream); line: 1, column: 13] (through reference chain: market.dto.CartItemDTO[\"_links\"])"))
                .body("'description'", containsString("uri=/customer/cart"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (500) PUT:/customer/contacts
    * Found 2 potential faults. Type-codes: 100, 200
    */
    @Test(timeout = 60000)
    public void test_4_with500() throws Exception {
        
        // Fault100. HTTP Status 500. PUT:/customer/contacts
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/contacts -> Response status 500 not defined for path '/customer/contacts'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" { " + 
                    " \"_links\": [ " + 
                    " { " + 
                    " \"deprecation\": \"Jb\", " + 
                    " \"href\": \"YiRBW\", " + 
                    " \"hreflang\": \"XqT\", " + 
                    " \"name\": \"AVfkdNngq\", " + 
                    " \"profile\": \"aDX175XiqArEzgc\", " + 
                    " \"type\": \"o6TU\" " + 
                    " }, " + 
                    " { " + 
                    " \"deprecation\": \"rWccVn79Lcc\", " + 
                    " \"hreflang\": \"nZMYyfsm4R9\", " + 
                    " \"media\": \"ddfn2yJ4\", " + 
                    " \"profile\": \"R7qiKS4aXKVZG\", " + 
                    " \"rel\": { " + 
                    " \"EM_tainted_map\": \"_EM_1_XYZ_\" " + 
                    " }, " + 
                    " \"title\": \"\" " + 
                    " }, " + 
                    " { " + 
                    " \"deprecation\": \"7ujMTokq\", " + 
                    " \"media\": \"F_1nY\", " + 
                    " \"profile\": \"k\", " + 
                    " \"type\": \"oe6X8zz3rryY\" " + 
                    " }, " + 
                    " { " + 
                    " \"href\": \"mKcdn9AZGeXaRB6v\", " + 
                    " \"hreflang\": \"_bvnDK8VVKVu\", " + 
                    " \"name\": \"IwMY\", " + 
                    " \"rel\": { " + 
                    " \"EM_tainted_map\": \"_EM_1_XYZ_\" " + 
                    " }, " + 
                    " \"title\": \"BT2ijtlOs7EF2Vm\", " + 
                    " \"type\": \"CqdgE4Q\" " + 
                    " } " + 
                    " ], " + 
                    " \"address\": \"ue\", " + 
                    " \"phone\": \"\" " + 
                    " } ")
                .put(baseUrlOfSut + "/customer/contacts?name=a05HXXFQtEEce")
                .then()
                .statusCode(500)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("JSON parse error: Expected relation name; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Expected relation name\n at [Source: (PushbackInputStream); line: 1, column: 13] (through reference chain: market.dto.ContactsDTO[\"_links\"])"))
                .body("'description'", containsString("uri=/customer/contacts"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (200) GET:/products
    */
    @Test(timeout = 60000)
    public void test_5() throws Exception {
        
        given().accept("*/*")
                .get(baseUrlOfSut + "/products")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (200) GET:/v2/api-docs/
    */
    @Test(timeout = 60000)
    public void test_6() throws Exception {
        
        given().accept("*/*")
                .get(baseUrlOfSut + "/v2/api-docs/")
                .then()
                .statusCode(200);
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer/orders
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_7() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer/orders -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer/orders?name=ZzMmX2y")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/orders"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer/contacts
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_8() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer/contacts -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer/contacts?name=PFjaakb")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/contacts"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) DELETE:/customer/cart
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_9() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. DELETE:/customer/cart -> No response body is expected but one was found.
        given().accept("*/*")
                .delete(baseUrlOfSut + "/customer/cart?name=15vWgEH7TsZNsK")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/cart"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer/cart
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_10() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer/cart -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer/cart")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/cart"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_11() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer?name=6HqDBQ5plotRyYoR")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'status'", numberMatches(401.0))
                .body("'error'", containsString("Unauthorized"))
                .body("'message'", containsString(""))
                .body("'path'", containsString("/customer"));
    }
    
    
    /**
    * Calls:
    * (401) DELETE:/customer/cart
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_12() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. DELETE:/customer/cart -> No response body is expected but one was found.
        given().accept("*/*")
                .delete(baseUrlOfSut + "/customer/cart")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/cart"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer/contacts
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_13() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer/contacts -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer/contacts")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/contacts"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer/cart
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_14() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer/cart -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer/cart?name=5z22yfGe")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/cart"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_15() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'status'", numberMatches(401.0))
                .body("'error'", containsString("Unauthorized"))
                .body("'message'", containsString(""))
                .body("'path'", containsString("/customer"));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer/orders
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_16() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer/orders -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer/orders")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/orders"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer/orders/{orderId}
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_17() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer/orders/{orderId} -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer/orders/8314916881610356334")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/orders/8314916881610356334"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer/orders/{orderId}
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_18() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer/orders/{orderId} -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer/orders/683?name=DlIP")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/orders/683"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) PUT:/customer/cart
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_19() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/cart -> No response body is expected but one was found.
        given().accept("*/*")
                .contentType("application/json")
                .body(" { " + 
                    " \"productId\": 409, " + 
                    " \"quantity\": 71 " + 
                    " } ")
                .put(baseUrlOfSut + "/customer/cart?name=WuE8Ta7tjJA")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/cart"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) GET:/customer/orders/{orderId}
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_20() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/customer/orders/{orderId} -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/customer/orders/-1183315444?name=p")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/orders/-1183315444"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) PUT:/customer/cart/delivery
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_21() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/cart/delivery -> No response body is expected but one was found.
        given().accept("*/*")
                .put(baseUrlOfSut + "/customer/cart/delivery?" + 
                    "included=false&" + 
                    "name=GBElFf4")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/cart/delivery"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) PUT:/customer/cart/delivery
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_22() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/cart/delivery -> No response body is expected but one was found.
        given().accept("*/*")
                .put(baseUrlOfSut + "/customer/cart/delivery?" + 
                    "included=true&" + 
                    "name=J")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/cart/delivery"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (401) PUT:/customer/cart/delivery
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_23() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/cart/delivery -> No response body is expected but one was found.
        given().accept("*/*")
                .put(baseUrlOfSut + "/customer/cart/delivery?included=true")
                .then()
                .statusCode(401)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Access is denied"))
                .body("'description'", containsString("uri=/customer/cart/delivery"))
                .body("'entityName'", nullValue())
                .body("'fieldErrors'.size()", equalTo(0));
    }
    
    
    /**
    * Calls:
    * (404) GET:/products/{productId}
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_24() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/products/{productId} -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/products/624")
                .then()
                .statusCode(404)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Requested entity doesn't exist"))
                .body("'description'", containsString("uri=/products/624"))
                .body("'entityName'", containsString("ProductDTO"))
                .body("'fieldErrors'.size()", equalTo(1))
                .body("'fieldErrors'[0].'field'", containsString("id"))
                .body("'fieldErrors'[0].'message'", containsString("No instance with this id"));
    }
    
    
    /**
    * Calls:
    * (404) GET:/products/{productId}
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_25() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. GET:/products/{productId} -> No response body is expected but one was found.
        given().accept("*/*")
                .get(baseUrlOfSut + "/products/-8791924785976515460")
                .then()
                .statusCode(404)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Requested entity doesn't exist"))
                .body("'description'", containsString("uri=/products/-8791924785976515460"))
                .body("'entityName'", containsString("ProductDTO"))
                .body("'fieldErrors'.size()", equalTo(1))
                .body("'fieldErrors'[0].'field'", containsString("id"))
                .body("'fieldErrors'[0].'message'", containsString("No instance with this id"));
    }
    
    
    /**
    * Calls:
    * (406) POST:/register
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_26() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. POST:/register -> Response status 406 not defined for path '/register'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" { " + 
                    " \"address\": \"b7VztzgVmjb5Nw\", " + 
                    " \"email\": \"Si_zNhapE0H3j8d\", " + 
                    " \"name\": \"-\\\\\\\\\", " + 
                    " \"phone\": \"\" " + 
                    " } ")
                .post(baseUrlOfSut + "/register")
                .then()
                .statusCode(406)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Argument validation error"))
                .body("'description'", containsString("uri=/register"))
                .body("'entityName'", containsString("userDTO"))
                .body("'fieldErrors'.size()", equalTo(4))
                .body("'fieldErrors'[0].'field'", containsString("email"))
                .body("'fieldErrors'[0].'message'", containsString("The value shall be in the format of an email address"))
                .body("'fieldErrors'[1].'field'", containsString("phone"))
                .body("'fieldErrors'[1].'message'", containsString("Shall consist of a + sign, a country code, a region code (1-4 digits), and a number (6-7 digits)"))
                .body("'fieldErrors'[2].'field'", containsString("phone"))
                .body("'fieldErrors'[2].'message'", containsString("The value shall not be empty"))
                ; // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
    }
    
    
    /**
    * Calls:
    * (406) POST:/customer/cart/pay
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_27() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. POST:/customer/cart/pay -> Response status 406 not defined for path '/customer/cart/pay'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" {} ")
                .post(baseUrlOfSut + "/customer/cart/pay")
                .then()
                .statusCode(406)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Argument validation error"))
                .body("'description'", containsString("uri=/customer/cart/pay"))
                .body("'entityName'", containsString("creditCardDTO"))
                .body("'fieldErrors'.size()", equalTo(1))
                .body("'fieldErrors'[0].'field'", containsString("ccNumber"))
                .body("'fieldErrors'[0].'message'", containsString("The value shall not be empty"));
    }
    
    
    /**
    * Calls:
    * (406) PUT:/customer/cart
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_28() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/cart -> Response status 406 not defined for path '/customer/cart'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" { " + 
                    " \"quantity\": 300 " + 
                    " } ")
                .put(baseUrlOfSut + "/customer/cart?name=ObbxCj0Q")
                .then()
                .statusCode(406)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Argument validation error"))
                .body("'description'", containsString("uri=/customer/cart"))
                .body("'entityName'", containsString("cartItemDTO"))
                .body("'fieldErrors'.size()", equalTo(1))
                .body("'fieldErrors'[0].'field'", containsString("productId"))
                .body("'fieldErrors'[0].'message'", containsString("Value shall be a positive number"));
    }
    
    
    /**
    * Calls:
    * (406) PUT:/customer/contacts
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_29() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. PUT:/customer/contacts -> Response status 406 not defined for path '/customer/contacts'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" { " + 
                    " \"address\": \"wTtR1ZVEH\", " + 
                    " \"phone\": \"\" " + 
                    " } ")
                .put(baseUrlOfSut + "/customer/contacts?name=x7AMfuiDn5fYue")
                .then()
                .statusCode(406)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Argument validation error"))
                .body("'description'", containsString("uri=/customer/contacts"))
                .body("'entityName'", containsString("contactsDTO"))
                .body("'fieldErrors'.size()", equalTo(2))
                .body("'fieldErrors'[0].'field'", containsString("phone"))
                .body("'fieldErrors'[0].'message'", containsString("The value shall not be empty"))
                .body("'fieldErrors'[1].'field'", containsString("phone"))
                .body("'fieldErrors'[1].'message'", containsString("Shall consist of a + sign, a country code, a region code (1-4 digits), and a number (6-7 digits)"));
    }
    
    
    /**
    * Calls:
    * (406) POST:/customer/cart/pay
    * Found 1 potential fault of type-code 200
    */
    @Test(timeout = 60000)
    public void test_30() throws Exception {
        
        // Fault200. Received A Response From API That Is Not Valid According To Its Schema. POST:/customer/cart/pay -> Response status 406 not defined for path '/customer/cart/pay'.
        given().accept("*/*")
                .contentType("application/json")
                .body(" {} ")
                .post(baseUrlOfSut + "/customer/cart/pay?name=9_")
                .then()
                .statusCode(406)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Argument validation error"))
                .body("'description'", containsString("uri=/customer/cart/pay"))
                .body("'entityName'", containsString("creditCardDTO"))
                .body("'fieldErrors'.size()", equalTo(1))
                .body("'fieldErrors'[0].'field'", containsString("ccNumber"))
                .body("'fieldErrors'[0].'message'", containsString("The value shall not be empty"));
    }


}
